src/compiler/types.ts

```diff
@@ -2555,7 +2558,13 @@ namespace ts {
     }
 
     /* @internal */
-    export type CommandLineOption = CommandLineOptionOfCustomType | CommandLineOptionOfPrimitiveType | TsConfigOnlyOption;
+    export interface CommandLineOptionOfListType extends CommandLineOptionBase {
+        type: "list";
+        element: CommandLineOptionOfCustomType | CommandLineOptionOfPrimitiveType;

```

Comment:
Here's an idea - create an interface called `CommandLineOptionTypeSpecifier` which `CommandLineOptionBase` extends from. Then use it here, and that way you won't have to specify the `name` for each array element.